File: Momento_2/Bloque_6/Animal/C++/src/Cat.cpp
#include "../include/Cat.h"

Cat::Cat() : Mammal() {}
Cat::Cat(std::string name) : Mammal(name) {}

std::string Cat::toString() const
{
    return "Cat [" + Mammal::toString() + " ]";
}

std::ostream &operator<<(std::ostream &os, const Cat &cat)
{
    os << cat.toString();
    return os;
}

File: Momento_2/Bloque_6/Animal/JAVA/src/Cat.java
public class Cat extends Mammal {

    private razaCat raza;

    public Cat() {
        super();
    }

    public Cat(String name, razaCat raza) {
        super(name);
        this.raza = raza;
    }

    public razaCat getRaza() {
        return raza;
    }

    public void setRaza(razaCat raza) {
        this.raza = raza;
    }

    @Override
    public String toString() {
        return "Cat [" + super.toString() + ", Raza: " + raza + "]";
    }

    public void greets() {
        System.out.println("Meow!");
    }
}
